hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieActiveTimeline.java

```diff
@@ -267,21 +267,27 @@ public void deleteCompactionRequested(HoodieInstant instant) {
   }
 
   public void deleteInstantFileIfExists(HoodieInstant instant) {
+    deleteInstantFileIfExists(instant, true);
+  }
+
+  public boolean deleteInstantFileIfExists(HoodieInstant instant, boolean exceptionIfFailToDelete) {
     LOG.info("Deleting instant " + instant);
-    Path inFlightCommitFilePath = getInstantFileNamePath(instant.getFileName());
+    Path commitFilePath = getInstantFileNamePath(instant.getFileName());
```

Comment:
we need to be careful about what commit can be deleted. this API is designed to only delete requested/inflight or empty clean commit, based on the usage. (1 exception is in `org.apache.hudi.cli.commands.TestRepairsCommand#testShowFailedCommits` where this api is misused for deleting completed commits in tests; we should make separate test helper for that)

We can't delete completed commit instants, which breaks the timeline's integrity. So we should name it properly at the variable as well as the API level.

Reply:
then we should add some doc for this method. `inFlightCommitFilePath` doesn't represent  requested and empty clean instant.